fuel-core/src/executor.rs

```diff
@@ -169,51 +169,72 @@ impl Executor {
                     }
                 }) {
                     TransactionStatus::Failed {
-                        block_id,
-                        time: block.headers.time,
+                        block_id: Default::default(),
+                        time: block.header.time,
                         reason: format!("{:?}", script_result.reason()),
                         result: Some(*vm_result.state()),
                     }
                 }
                 // otherwise just log the revert arg
                 else {
                     TransactionStatus::Failed {
-                        block_id,
-                        time: block.headers.time,
+                        block_id: Default::default(),
+                        time: block.header.time,
                         reason: format!("{:?}", vm_result.state()),
                         result: Some(*vm_result.state()),
                     }
                 }
             } else {
                 // else tx was a success
                 TransactionStatus::Success {
-                    block_id,
-                    time: block.headers.time,
+                    block_id: Default::default(),
+                    time: block.header.time,
                     result: *vm_result.state(),
                 }
             };
 
-            // persist tx status at the block level
-            block_db_transaction.update_tx_status(&tx_id, status)?;
+            // queue up status for this tx to be stored once block id is finalized.
+            tx_status.push((tx_id, status));
         }
 
         // check or set transaction commitment
+        let txs_root = txs_merkle
+            .root()
+            .expect("In-memory impl should be infallible")
+            .into();
         match mode {
             ExecutionMode::Production => {
-                block.headers.transactions_commitment = commitment;
+                block.header.transactions_root = txs_root;
             }
             ExecutionMode::Validation => {
-                if block.headers.transactions_commitment != commitment {
+                if block.header.transactions_root != txs_root {
                     return Err(Error::InvalidBlockCommitment);
                 }
             }
         }
+        // set the coinbase amount
+        block.header.coinbase = coinbase;
```

Comment:
This probably still needs to be defines, but i think Nick mentioned long ago that fuel is going to be deflationary so in that sense `coinbase` needs to be increase by some fixed amount to reward validator and we will need to check under `ExecutionMode::Validation` if that amount is okay or not.

Reply:
Good catch

Reply:
Although it may still be useful to capture the total fees of the block, even if it doesn't all go directly to the validator and some get burned.

Reply:
I've been thinking of having a third coinbase tx type that's 0-input, 1-output. That way there's an explicit TXO for the coinbase rather than some implicit state mutation like with Ethereum.

Reply:
> I've been thinking of having a third coinbase tx type that's 0-input, 1-output. That way there's an explicit TXO for the coinbase rather than some implicit state mutation like with Ethereum.

This is similar problem with token deposits, we need to make token out of thin air in fuel :). Maybe having additional Input that will cover those two cases?

Reply:
There's an issue to track deposit IDs here: https://github.com/FuelLabs/fuel-specs/issues/106

Reply:
I've removed the coinbase field and will just `debug` log the fee amounts for now.